lib/SimpleSAML/Module.php

```diff
@@ -265,11 +265,21 @@ public static function process(Request $request = null)
         }
 
         $response = new BinaryFileResponse($path);
-        $response->setCache(['public' => true, 'max_age' => 86400]);
-        $response->setExpires(new \DateTime(gmdate('D, j M Y H:i:s \G\M\T', time() + 10 * 60)));
-        $response->setLastModified(new \DateTime(gmdate('D, j M Y H:i:s \G\M\T', filemtime($path))));
+        $cacheSettings = $config->getArray('module.php.cache', ['public' => true, 'max_age' => 86400]);
```

Comment:
Here we could just get the `caching` config group, process it to validate the supported options, and then go ahead acting accordingly for each of them. Those that cannot be passed to `setCache()` could be processed first and removed from the array, and then the remaining could be passed to `setCache()` directly. It might be necessary to have some validation in place, although I believe Symfony would handle that for us.